syntax = "proto3";

option go_package = "github.com/8thgencore/mailfort/proto";

package mail.v1;

// The request message containing the email and otp code.
message SendEmailWithOTPCodeRequest {
    string email = 1;
    string otp_code = 2;
}

// The response message.
message Response {
    string message = 1;
    bool success = 2;
}

// The error response message.
message ErrorResponse {
    repeated string messages = 1;
    bool success = 2;
}

service MailService {
    // Sends a confirmation email otp code.
    rpc SendConfirmationEmailOTPCode(SendEmailWithOTPCodeRequest) returns (Response);

    // Sends a password reset otp code.
    rpc SendPasswordResetOTPCode(SendEmailWithOTPCodeRequest) returns (Response);
}
